### ISSUE TITLE
Issue with enum initiated with Array<Something> type

### ISSUE THREAD

**Issue by jakubwolny** [colinhacks/zod#3080]
I'm trying to migrate quite big types definitions library (100+ files) from superstruct to zod for following reasons
- readonly()
- discriminatedUnion()

So far it's been smooth ride I'm having issue with "enum" type. We mostly used 2 patterns in superstruct:
```
// (1) real enum
enum SomethingType {
  SOMETHING = 'SOMETHING'
}

// (2) fake enum 
const SomethingType = {
  SOMETHING: 'SOMETHING'
} as const

// both were used fine in superstruct enum - by simply converting then into Array of something
const sortedValues = Object.values(SomethingType) // Array<"SOMETHING">
const superstructEnum = enums(sortedValues)

export const zodEnum = z.enum(SomethingType) // error Argument of type SomethingType[] is not assignable to parameter of type readonly [string, ...string[]]
```

Why we used fake enum is a bit out of scope of this question, but if you're curious: for migrating between the versions where enum members change. The issue I'm having with zod is that none of these ways of defining enums seem to work. I guess it's all because of the requirement of having first element in the tuple:

`createZodEnum<U extends string, T extends Readonly<[U, ...U[]]>>`

can I ask - what's the reason or benefit of having this?

Also, I know I probably could change from enum() to union() but for this I need to define all these strings as literals, which is quite big change and affects the library's public API. Is there any other way to define the enum in zod, which is similar to the one in superstruct?

Would this PR https://github.com/colinhacks/zod/pull/2338 fix my problem?

**Comment by JacobWeisenburger** [comment]
Have you had a chance to read thru the docs for Zod Enums?
https://github.com/colinhacks/zod#zod-enums

**Comment by jakubwolny** [comment]
hi @JacobWeisenburger 
thank you for your answer - unfortunately, it doesn't work for me. To be specific:
```
z.enum(['SOMETHING', 'SOMETHING_ELSE']) // works fine, no issues here

const items = {
  SOMETHING: 'SOMETHING', 
  SOMETHING_ELSE: 'SOMETHING_ELSE'
} as const

z.enum(Object.values(items)) // does NOT work, even though it's effectively the same
// TS error: Argument of type string[] is not assignable to parameter of type [string, ...string[]]
```
so I'm guessing the issue is that "z.enum" requires a tuple (with at least 1 element, instead of an array). I could of course get around this problem with forcing this this, but my question was (a) why is that required? and (b) would aforementioned PR fix it?

**Comment by JacobWeisenburger** [comment]
Does this work for you?

```ts
type Options = keyof typeof items
const items = {
    SOMETHING: 'SOMETHING',
    SOMETHING_ELSE: 'SOMETHING_ELSE'
} as const

z.enum( Object.values( items ) as [ Options, ...Options[] ] )
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by jakubwolny** [comment]
Yes, that's one possible workaround - just I don't see any reason why this is required.

**Comment by JacobWeisenburger** [comment]
if all you are wanting with this issue is to find out why, then can I move it to the discussions? Or are you asking for it to be changed?
